src/actions/summit-event-bulk-actions.js

```diff
@@ -48,35 +49,79 @@ export const UPDATE_STREAMING_TYPE_BULK       = 'UPDATE_STREAMING_TYPE_BULK';
 export const UPDATE_MEETING_URL_BULK          = 'UPDATE_MEETING_URL_BULK';
 export const UPDATE_ETHERPAD_URL_BULK         = 'UPDATE_ETHERPAD_URL_BULK';
 
-export const getSummitEventsById = (summitId, eventIds ) => async (dispatch, getState) => {
-
+export const getSummitEventsById = (events) => async (dispatch, getState) => {
     const { currentSummitState } = getState();
     const accessToken = await getAccessTokenSafely();
     const { currentSummit }   = currentSummitState;
+    const filter = [`id==${events.join('||')}`]
+
     dispatch(startLoading());
-    let filter = '';
-    for(let id of eventIds){
-        if(filter !== '') filter += ',';
-        filter += `id==${id}`;
-    }
 
     const params = {
         access_token : accessToken,
-        filter: filter,
+        'filter[]': filter,
         page: 1,
-        per_page: eventIds.length < 5 ? 5 : eventIds.length
+        per_page: 200
     };
 
     return getRequest(
-        null,
-        createAction(RECEIVE_SELECTED_EVENTS),
-        `${window.API_BASE_URL}/api/v1/summits/${summitId}/events`,
-        authErrorHandler
+      createAction(REQUEST_SELECTED_EVENTS),
+      createAction(RECEIVE_SELECTED_EVENTS),
+      `${window.API_BASE_URL}/api/v1/summits/${currentSummit.id}/events`,
+      authErrorHandler
     )(params)(dispatch).then(() => {
-            dispatch(stopLoading());
-            dispatch(createAction(UPDATE_VALIDATION_STATE)({currentSummit}));
-        }
-    );
+        dispatch(stopLoading());
+        dispatch(createAction(UPDATE_VALIDATION_STATE)({currentSummit}));
+    })
+};
+
+export const getSummitEventsByFilters = () => async (dispatch, getState) => {
+    const { currentSummitState, summitEventsBulkActionsState } = getState();
+    const accessToken = await getAccessTokenSafely();
+    const { currentSummit }   = currentSummitState;
+    const {
+        selectedAllUnPublished, selectedUnPublishedEvents, excludedUnPublishedEvents, unPublishedFilter, totalUnPublished
+    } = summitEventsBulkActionsState
+
+    dispatch(startLoading());
+
+    const lastPage = Math.ceil(totalUnPublished / 200);
```

Comment:
here code is assuming that the lastPage is over the total
but what about if u filter and then select a bunch by hand ?
then code will be doing more request than needed if the  selectedUnPublishedEvents.length > 0